### PR TITLE
Add support for connection_pool gem

### PR DESCRIPTION
**Pull request by danila**
Adding support for @mperham's [connection_pool](https://github.com/mperham/connection_pool), in order to make gems compatible without the need to rely on method_missing that is used in [wrapper](https://github.com/mperham/connection_pool#migrating-to-a-connection-pool).

This PR was influenced by https://github.com/resque/redis-namespace/pull/135. Unfortunately we can't rely on `respond_to?(:with)` anymore, as this method was introduced to `redis-rb` gem [here](https://github.com/redis/redis-rb/commit/674541c5cc162b46066556811967a28a184f4665#), so we gotta check class name.

### CODE REVIEW

File: lib/redis/namespace.rb [hunk 1]
```diff
@@ -372,7 +372,14 @@ def method_missing(command, *args, &block)
                "passthrough has been deprecated and will be removed in " +
                "redis-namespace 2.0 (at #{call_site})")
         end
-        @redis.send(command, *args, &block)
+
+        if @redis.class.name == "ConnectionPool"
+          @redis.with do |pool_connection|
```
Comment (iloveitaly) [inline lib/redis/namespace.rb:377]: I'm not familiar with connection pool: is there a way to get the `pool_connection` without using `with`?

i.e. we could do something like:

```ruby
redis_connection = if @redis.class.name == "ConnectionPool"
  @redis.connection_pool
else
  @redis
end
```

And then have a single `send` call?

File: spec/redis_spec.rb [hunk 2]
```diff
@@ -403,6 +449,21 @@
     expect(@namespaced.hgetall("foo")).to eq({"key1" => "value1"})
   end
 
+  it "should utilize connection_pool while adding namespace to pipelined blocks" do
```
Comment (iloveitaly) [inline spec/redis_spec.rb:452]: how do we know connection pool is being used in this test? Can we assert something on the connection pool instance?

**Review by iloveitaly — CHANGES_REQUESTED** [review]
Awesome contribution! A couple questions + improvements before i merge this in.

File: spec/redis_spec.rb (hunk 2 above)
Comment (danila) [reply spec/redis_spec.rb:467]: Added assertions on ConnectionPool calls 👍

**Comment by danila** [comment]
Hey, @iloveitaly thank you for the fastest response! Indeed the initial approach was dirty so I refactored it a bit and extended the specs.

File: lib/redis/namespace.rb [hunk 3]
```diff
@@ -531,12 +545,28 @@ def call_site
...
-          copy = dup
-          copy.redis = r
-          yield copy
+        if redis.class.name == "ConnectionPool"
+          redis.with do |pool_connection|
+            pool_connection.send(command) do |r|
+              copy = dup
+              copy.redis = r
+              yield copy
+            end
```
[the remark this answers is not in the document]
Comment (iloveitaly) [reply lib/redis/namespace.rb]: Figured!

File: lib/redis/namespace.rb (hunk 1 above)
Comment (iloveitaly) [reply lib/redis/namespace.rb]: Figured, makes sense!

**Pull request merged**